tests/environment-hook.bats

```diff
@@ -29,25 +31,22 @@ load '/usr/local/lib/bats/load.bash'
   refute_output --partial "ANOTHER_SECRET=baa"
 
   unstub vault
-
-  unset TESTDATA
-  unset BUILDKITE_PLUGIN_VAULT_SECRETS_DUMP_ENV
-  unset BUILDKITE_PLUGIN_VAULT_SECRETS_SERVER
-  unset BUILDKITE_PIPELINE_SLUG
 }
 
 @test "Load default environment file from vault server" {
   export BUILDKITE_PLUGIN_VAULT_SECRETS_SERVER=https://vault_svr_url
   export BUILDKITE_PLUGIN_VAULT_SECRETS_DUMP_ENV=true
+  export TESTDATA='MY_SECRET=fooblah'
   export BUILDKITE_PIPELINE_SLUG=testpipe
-  export TESTDATA=`echo MY_SECRET=fooblah | base64`
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_SECRET_ENV="abcde12345"
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_ROLE_ID=buildkite
+  export BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_METHOD=approle
 
   stub vault \
-    'auth -address=https://vault_svr_url - : echo Successfully authenticated. You are now logged in' \
-    "list -address=https://vault_svr_url -format=yaml data/buildkite/testpipe : exit 0 "\
-    "list -address=https://vault_svr_url -format=yaml data/buildkite : echo -e '- environment'" \
-    "list -address=https://vault_svr_url -format=yaml data/buildkite/environment : echo -e '- MY_SECRET'" \
-    "read -address=https://vault_svr_url -field=value data/buildkite/environment/MY_SECRET : echo ${TESTDATA}"
+    "write -field=token -address=https://vault_svr_url auth/approle/login role_id=buildkite secret_id=\"${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_SECRET_ENV}\" : echo 'Successfully authenticated with Role ID ${BUILDKITE_PLUGIN_VAULT_SECRETS_AUTH_ROLE_ID}'" \
```

Comment:
you can reuse the same generic stub as before to make it more general and less verbose:

```suggestion
    "write -field=token \* auth/approle/login \* \* : echo 'Successfully authenticated with \${5} \${6}" \
```

Reply:
Both this and the other tests which had an auth method [configured ](https://github.com/buildkite-plugins/vault-secrets-buildkite-plugin/pull/13#discussion_r1111105758) didn't actually need them to be defined to test the functionality, so I've removed this stub